vulcanai2/models/cnn.py

```diff
@@ -92,8 +94,42 @@ def __init__(self, name, in_dim, config, save_path=None,
             lr_scheduler, early_stopping, criter_spec)
 
     def _create_network(self, **kwargs):
-
+        self._in_dim = self.in_dim
         conv_hid_layers = self._config.units
+
+        if len(self.in_dim) > 1:
```

Comment:
you have to be very careful with this conditional. If for some reason it's not a list of tuples than a simple convnet with dim (1, 28, 28) will think it's 3 separate inputs